src/components/scenes/CoinRankingScene.tsx

```diff
@@ -204,47 +206,75 @@ const CoinRankingComponent = (props: Props) => {
-            for (let i = 0; i < listings.data.length; i++) {
-              const rankIndex = start - 1 + i
-              const row = listings.data[i]
-              coinRankingDatas[rankIndex] = row
-              debugLog(LOG_COINRANK, `queryLoop: ${rankIndex.toString()} ${row.rank} ${row.currencyCode}`)
-            }
-            start += QUERY_PAGE_SIZE
+  React.useEffect(() => {
+    // Start querying starting from either the last fetched index (scrolling) or
+    // the first index (initial load/timed refresh)
+    const queryLoop = async (start = lastStartIndex) => {
+      debugLog(LOG_COINRANK, `queryLoop(start: ${start})`)
+
+      if (isQuerying) {
+        debugLog(LOG_COINRANK, '** Skipping query **')
+        return
+      }
+      isQuerying = true
+      try {
+        // Catch up to the total required items
+        while (start < requestDataSize - QUERY_PAGE_SIZE) {
+          const url = `v2/coinrank?fiatCode=iso:${supportedFiatSetting}&start=${start}&length=${QUERY_PAGE_SIZE}`
+
+          const response = await fetchRates(url)
+          if (!response.ok) {
+            const text = await response.text()
+            console.warn(`API call failed with response: ${text}`)
+            break
           }
-          setDataSize(coinRankingDatas.length)
-          if (lastFetchedFiat !== supportedFiatSetting) {
-            setLastFetchedFiat(supportedFiatSetting)
+          const replyJson = await response.json()
+          const listings = asCoinranking(replyJson)
+          for (let i = 0; i < listings.data.length; i++) {
+            const rankIndex = start - 1 + i
+            const row = listings.data[i]
+            coinRankingDatas[rankIndex] = row
+            debugLog(LOG_COINRANK, `queryLoop: ${rankIndex.toString()} ${row.rank} ${row.currencyCode}`)
           }
-        } catch (e: any) {
-          console.warn(e.message)
+          start += QUERY_PAGE_SIZE
         }
-        timeoutHandler.current = setTimeout(queryLoop, LISTINGS_REFRESH_INTERVAL)
+      } catch (e: any) {
+        console.warn(`Error during data fetch: ${e.message}`)
       }
-      if (timeoutHandler.current != null) {
+
+      setDataSize(coinRankingDatas.length)
+      setLastStartIndex(start)
+      isQuerying = false
+
+      // Refresh from the beginning periodically
+      timeoutHandler.current = setTimeout(async () => {
+        debugLog(LOG_COINRANK, 'Refreshing list')
+        await queryLoop(1)
+      }, LISTINGS_REFRESH_INTERVAL)
+    }
+
+    if (lastFetchedFiat !== supportedFiatSetting) {
+      // Reset everything when the fiat setting does not match what cached
+      // data we fetched previously.
+      debugLog(LOG_COINRANK, `Fiat changed from ${lastFetchedFiat} to ${supportedFiatSetting}`)
+      isQuerying = false
+      if (timeoutHandler.current) clearTimeout(timeoutHandler.current)
+      coinRanking.coinRankingDatas = []
+      setLastStartIndex(1)
+      setDataSize(0)
+      setRequestDataSize(QUERY_PAGE_SIZE)
```

Comment:
Need to trigger a query loop if fiat changes. Just take the else case and make it always run